File: ProblemSolving/450DSA/Python/src/bitmanipulation/CountSetBits.py
"""
@author Anirudh Sharma

Given a positive integer N, print count of set bits in it.
"""


def countSetBits(N):
    # Count of set bits
    count = 0
    # Loop until N becomes zero
    while N > 0:
        N &= (N - 1)
        count += 1
    return count


if __name__ == "__main__":
    print(countSetBits(6))
    print(countSetBits(31))
    print(countSetBits(13))
    print(countSetBits(40))

File: ProblemSolving/450DSA/JavaScript/src/bitmanipulation/CountSetBits.js
/**
 * @author Anirudh Sharma
 * 
 * Given a positive integer N, print count of set bits in it.
 */
const countSetBits = (N) => {
    // Count of set bits
    let count = 0;
    // Loop until N becomes zero
    while (N > 0) {
        N &= (N - 1);
        count++;
    }
    return count;
};

const main = () => {
    console.log(countSetBits(6));
    console.log(countSetBits(31));
    console.log(countSetBits(13));
    console.log(countSetBits(40));
};

main();